### PR TITLE
Bedrock shield Implementation

### PR DESCRIPTION
**Pull request by rsgrewal-aws**
adding classes for bedrock shield. Draft need these to be reviewed please

### CODE REVIEW

**Comment by ashwinb** [comment]
@rsgrewal-aws thanks for the PR! Looking at this now

File: llama_stack/providers/adapters/safety/bedrock/config.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+from llama_models.schema_utils import json_schema_type
+from pydantic import BaseModel, Field
+import boto3
+
+
+@json_schema_type
+class BedrockShieldConfig(BaseModel):
+    """Configuration information for a guardrail that you want to use in the request."""
+
+    boto_client: Optional[boto3] = Field(
```
Comment (ashwinb) [inline llama_stack/providers/adapters/safety/bedrock/config.py:18]: the config is a data object, no code object should be here. you should instantiate the boto client in the initialize method of your adapter -- or if the client needs API keys then only at the time of the request when you can read these keys.

File: llama_stack/providers/adapters/safety/bedrock/config.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        default=None,
+        description="The client being used to connect to bedrock. This will ensure separation of creation to invocation",
+    )
+
+    gr_name: Optional[str] = Field(
```
Comment (ashwinb) [inline llama_stack/providers/adapters/safety/bedrock/config.py:23]: is this field fixed by the server or is this something you expect the client to specify for each request?

File: llama_stack/providers/adapters/safety/bedrock/bedrock.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        except Exception as e:
+            import traceback
+
+            traceback.print_exc()
+            raise RuntimeError(f"Error initializing BedrockSafetyAdapter: {e}") from e
+
+    async def shutdown(self) -> None:
+        pass
+
+    async def completion(self, request: CompletionRequest) -> AsyncGenerator:
```
Comment (ashwinb) [inline llama_stack/providers/adapters/safety/bedrock/bedrock.py:46]: this method is irrelevant for the `Safety` interface

**Comment by rsgrewal-aws** [comment]
Fixed the changes as disscussed and screen shot for successful run is attached

**Comment by rsgrewal-aws** [comment]
Screenshot 2024-09-24 at 6 32 49 PM

File: llama_stack/providers/adapters/safety/bedrock/config.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+# All rights reserved.
+#
+# This source code is licensed under the terms described in the LICENSE file in
+# the root directory of this source tree.
+
+from typing import Optional
+
+from llama_models.schema_utils import json_schema_type
+from pydantic import BaseModel, Field
+import boto3
```
Comment (ashwinb) [inline llama_stack/providers/adapters/safety/bedrock/config.py:11]: this is not needed here

**Review by ashwinb — APPROVED** [review]

**Pull request closed without merging**